### ISSUE TITLE
Support Fargate Profiles for AWS EKS

### ISSUE THREAD

**Issue by lanmalkieri** [terraform-providers/terraform-provider-aws#11109]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS recently launched support for Fargate in EKS: https://aws.amazon.com/blogs/aws/amazon-eks-on-aws-fargate-now-generally-available/

Would be amazing to have this supported quickly in Terraform!

### New or Affected Resource(s)

* aws_eks_cluster

### Potential Terraform Configuration

```hcl
resource "aws_eks_cluster" "example" {
  name     = "example"
  role_arn = aws_iam_role.example.arn

  vpc_config {
    subnet_ids = [aws_subnet.example1.id, aws_subnet.example2.id]
  }

 fargate_profile {
   name = "profile_name"
   subnets = subnet-xxxxxx
   # match namespaces
   namespaces []
   # match labels for pod creation
   labels = []
   pod_execution_role_arn = arn:xxxxx
}

```

### References
https://aws.amazon.com/blogs/aws/amazon-eks-on-aws-fargate-now-generally-available/

**Comment by bflad** [comment]
Please see: #11110

**Comment by bflad** [comment]
A new `aws_eks_fargate_profile` resource has been merged and will release with version 2.41.0 of the Terraform AWS Provider, later today or tomorrow.
